### DISCUSSION TITLE
What is the default value if did not specify the max_latency?

### DISCUSSION THREAD

**Discussion by junneng-my** [axboe/fio#1670] [Q&A] [answered]
I could not find the answer from here https://fio.readthedocs.io/en/latest/fio_doc.html?highlight=max%20latency#cmdoption-arg-max-latencyhttps://fio.readthedocs.io/en/latest/fio_doc.html?highlight=max%20latency#cmdoption-arg-max-latency

**Comment by bolochavaez** [comment, marked as answer]
There is no default defined in the code.

**Reply by junneng-my** [reply]
Hi @bolochavaez , thanks for the answer. Can I assume if max_latency value not given, IO command time-out will not throw error in FIO?

**Reply by bolochavaez** [reply]
I have ran some tests on hardware that resulted in some pretty high max response times without issues. How much latency are you expecting?

**Reply by sitsofe** [reply]
@junneng-my Be aware that [`max_latency`](https://fio.readthedocs.io/en/latest/fio_doc.html#cmdoption-arg-max_latency) only takes effect _after_ the I/O has returned - while the I/O is in-flight, fio will wait forever for it to return. After the I/O has returned if `max_latency` is set then the I/O's latency is checked against it and as @bolochavaez stated, if it's unset then no checking is done (and no errors will be thrown).

Obviously if the OS itself throws an error due to an I/O taking too long there's nothing fio can do to influence that (and fio will just see the I/O returning with an error).
